modules/reindex/src/test/java/org/elasticsearch/index/reindex/ReindexResilientSearchIT.java

```diff
@@ -0,0 +1,138 @@
+        assertAcked(prepareCreate("dest")
+            .addMapping("_doc", jsonBuilder()
+                .startObject()
+                    .startObject("properties")
+                        .startObject("data")
+                            .field("type", "long")
+                        .endObject()
+                    .endObject()
+                .endObject())
+            .setSettings(Settings.builder()
+                .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, shardCount)
+                .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 1)
+                .build()));
+
+        int numberOfDocuments = randomIntBetween(10, 50);
+        indexRandom(true, IntStream.range(0, numberOfDocuments)
+            .mapToObj(i -> client().prepareIndex("test", "doc", String.valueOf(i)).setSource("data", i)).collect(Collectors.toList()));
+
+        ensureGreen("test");
+        String reindexNode = internalCluster().startCoordinatingOnlyNode(Settings.EMPTY);
+        NodeClient reindexNodeClient = internalCluster().getInstance(NodeClient.class, reindexNode);
+
+
+        ReindexRequest request = new ReindexRequest();
+        request.setSourceIndices("test").setSourceBatchSize(1);
+        request.setDestIndex("dest");
+        request.setRequestsPerSecond(0.3f); // 30 seconds minimum
+        request.setAbortOnVersionConflict(false);
+        PlainListenableActionFuture<BulkByScrollResponse> reindexFuture = PlainListenableActionFuture.newListenableFuture();
+        Task reindexTask = reindexNodeClient.executeLocally(ReindexAction.INSTANCE, request, reindexFuture);
+
+        assertBusy(() -> {
+            IndicesStatsResponse indicesStatsResponse = client().admin().indices().prepareStats("test").execute().actionGet();
+            assertThat(Stream.of(indicesStatsResponse.getIndex("test").getShards())
+                    .mapToLong(shardStat -> shardStat.getStats().search.getTotal().getScrollCurrent()).sum(),
+                Matchers.equalTo((long) shardCount));
+        }, 30, TimeUnit.SECONDS);
+
+        for (int i = 0; i < randomIntBetween(1,5); ++i) {
+            internalCluster().restartRandomDataNode();
+        }
+
+        rethrottle().setTaskId(new TaskId(reindexNodeClient.getLocalNodeId(), reindexTask.getId()))
+            .setRequestsPerSecond(Float.POSITIVE_INFINITY).execute().get();
+
+        BulkByScrollResponse bulkByScrollResponse = reindexFuture.actionGet(30, TimeUnit.SECONDS);
+        assertThat(bulkByScrollResponse.getBulkFailures(), Matchers.empty());
+//        assertEquals(0, bulkByScrollResponse.getSearchFailures().size());
+
+        assertSameDocs(numberOfDocuments, "test", "dest");
+    }
+
+    private void assertSameDocs(int numberOfDocuments, String... indices) {
+        refresh(indices);
+        SearchSourceBuilder sourceBuilder = SearchSourceBuilder.searchSource().size(0)
```

Comment:
perhaps just do a count or use a cardinality aggregation on the "data" field? No need for scripting I think.